### PR TITLE
M3: polarity, interpolation, pump probe

### PR DESCRIPTION
**Pull request by thattil**

### CODE REVIEW

File: python/slsdet/detector.py [hunk 1]
```diff
@@ -3498,10 +3498,57 @@ def clkfreq(self):
...
     def readout(self):
         """
-        Mythen3] Starts detector readout. Status changes to TRANSMITTING and automatically returns to idle at the end of readout.
+        [Mythen3] Starts detector readout. Status changes to TRANSMITTING and automatically returns to idle at the end of readout.
         """
         self.startDetectorReadout()
     
+    @property
+    @element
+    def polarity(self):
```
Comment (erikfrojdh) [inline python/slsdet/detector.py:3507]: Polarity enum is missing from python. You can get it by running python generate_enums.py from the same folder as the function script.

File: slsDetectorServers/mythen3DetectorServer/mythen3.c [hunk 2]
```diff
@@ -149,54 +113,125 @@ patternParameters *setChipStatusRegisterPattern(int csr) {
...
+        csr |= 1 << _CSR_C15pre;
+
+    return setChipStatusRegister(csr);
+}
+
+int getInterpolation() {
+    return ((chipStatusRegister & CSR_interp_MSK) >> CSR_interp);
+}
+
+int setInterpolation(int mask) {
```
Comment (erikfrojdh) [inline slsDetectorServers/mythen3DetectorServer/mythen3.c:161]: Can we come up with a better name than mask?

**Review by erikfrojdh — CHANGES_REQUESTED** [review]
Few comment. Will have a second look when the I can test with python.

File: python/slsdet/detector.py (hunk 1 above)
Comment (thattil) [reply python/slsdet/detector.py:3507]: done

File: python/slsdet/detector.py [hunk 3]
```diff
@@ -3498,10 +3498,57 @@ def clkfreq(self):
...
+        """[Mythen3] Set positive or negative polarity. Enum: polarity"""
+        return self.getPolarity()
+
+    @polarity.setter
+    def polarity(self, value):
+        ut.set_using_dict(self.setPolarity, value)
+
+    @element
```
[the remark this answers is not in the document]
Comment (thattil) [reply python/slsdet/detector.py:3515]: done

File: slsDetectorServers/mythen3DetectorServer/mythen3.c (hunk 2 above)
Comment (thattil) [reply slsDetectorServers/mythen3DetectorServer/mythen3.c:161]: done

**Pull request merged**